test/networkd-test.py

```diff
@@ -726,6 +766,83 @@ def get_tz():
             raise
 
 
+class UnmanagedClientTest(unittest.TestCase, NetworkdTestingUtilities):
+    """Test if networkd manages the correct interfaces."""
+
+    def setUp(self):
+        """Write .network files to match the named veth devices."""
+        # Define the veth+peer pairs to be created.
+        # Their pairing doesn't actually matter, only their names do.
+        self.veths = {
+            'm1def': 'm0unm',
+            'm1man': 'm1unm',
+        }
+
+        # Define the contents of .network files to be read in order.
+        self.configs = (
+            "[Match]\nName=m1def\n",
+            "[Match]\nName=m1unm\n[Link]\nUnmanaged=yes\n",
+            "[Match]\nName=m1*\n[Link]\nUnmanaged=no\n",
+        )
+
+        # Write out the .network files to be cleaned up automatically.
+        for i in range(len(self.configs)):
+            self.write_network_unit("%02d-test.network" % i, self.configs[i])
+
+    def tearDown(self):
+        """Delete the test veth devices, and stop networkd."""
+        for peer in self.veths.values():
+            subprocess.check_call(['ip', 'link', 'del', 'dev', peer])
+        subprocess.call(['systemctl', 'stop', 'systemd-networkd'])
+
+    def create_iface(self):
+        """Create some veth pairs to use for interface matching."""
+        for veth in self.veths:
+            subprocess.check_call(['ip', 'link', 'add',
+                                   'name', veth, 'type', 'veth',
+                                   'peer', 'name', self.veths[veth]])
```

Comment:
It's a bit more compact and easier to see what happens to put a `self.addCleanup(subprocess.call, [..stop..]` here and drop it from `tearDown()`.